src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -233,9 +233,12 @@ internal void RemoveProject(AbstractProject project)
 
             UpdateProjectBinPath(project, project.TryGetBinOutputPath(), null);
 
-            if (project.PushingChangesToWorkspaceHosts)
+            using (Dispatcher.CurrentDispatcher.DisableProcessing())
             {
-                NotifyWorkspaceHosts(host => host.OnProjectRemoved(project.Id));
+                foreach (var hostState in _workspaceHosts)
```

Comment:
Won't it call into http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Workspace.cs,414 and throw because the project isn't in the solution?  Also, what about the CodeLens `WorkspaceHost`.